src/login/connect_server.h

```diff
@@ -0,0 +1,2257 @@
+    }
+
+    if (settings::get<bool>("login.CHAINS_OF_PROMATHIA"))
+    {
+        mask |= EXPANSION_DISPLAY::CHAINS_OF_PROMATHIA;
+    }
+
+    if (settings::get<bool>("login.TREASURES_OF_AHT_URGHAN"))
+    {
+        mask |= EXPANSION_DISPLAY::TREASURES_OF_AHT_URGHAN;
+    }
+
+    if (settings::get<bool>("login.WINGS_OF_THE_GODDESS"))
+    {
+        mask |= EXPANSION_DISPLAY::WINGS_OF_THE_GODDESS;
+    }
+
+    if (settings::get<bool>("login.A_CRYSTALLINE_PROPHECY"))
+    {
+        mask |= EXPANSION_DISPLAY::A_CRYSTALLINE_PROPHECY;
+    }
+
+    if (settings::get<bool>("login.A_MOOGLE_KUPOD_ETAT"))
+    {
+        mask |= EXPANSION_DISPLAY::A_MOOGLE_KUPOD_ETAT;
+    }
+
+    if (settings::get<bool>("login.A_SHANTOTTO_ASCENSION"))
+    {
+        mask |= EXPANSION_DISPLAY::A_SHANTOTTO_ASCENSION;
+    }
+
+    if (settings::get<bool>("login.VISIONS_OF_ABYSSEA"))
+    {
+        mask |= EXPANSION_DISPLAY::VISIONS_OF_ABYSSEA;
+    }
+
+    if (settings::get<bool>("login.SCARS_OF_ABYSSEA"))
+    {
+        mask |= EXPANSION_DISPLAY::SCARS_OF_ABYSSEA;
+    }
+
+    if (settings::get<bool>("login.HEROES_OF_ABYSSEA"))
+    {
+        mask |= EXPANSION_DISPLAY::HEROES_OF_ABYSSEA;
+    }
+
+    if (settings::get<bool>("login.SEEKERS_OF_ADOULIN"))
+    {
+        mask |= EXPANSION_DISPLAY::SEEKERS_OF_ADOULIN;
+    }
+    return mask;
+}
+
+uint16 generateFeatureBitmask()
+{
+    uint16 mask = 0;
+
+    if (settings::get<bool>("login.SECURE_TOKEN"))
+    {
+        mask |= FEATURE_DISPLAY::SECURE_TOKEN;
+    }
+
+    if (settings::get<bool>("login.MOG_WARDROBE_3"))
+    {
+        mask |= FEATURE_DISPLAY::MOG_WARDROBE_3;
+    }
+
+    if (settings::get<bool>("login.MOG_WARDROBE_4"))
+    {
+        mask |= FEATURE_DISPLAY::MOG_WARDROBE_4;
+    }
+
+    if (settings::get<bool>("login.MOG_WARDROBE_5"))
+    {
+        mask |= FEATURE_DISPLAY::MOG_WARDROBE_5;
+    }
+
+    if (settings::get<bool>("login.MOG_WARDROBE_6"))
+    {
+        mask |= FEATURE_DISPLAY::MOG_WARDROBE_6;
+    }
+
+    if (settings::get<bool>("login.MOG_WARDROBE_7"))
+    {
+        mask |= FEATURE_DISPLAY::MOG_WARDROBE_7;
+    }
+
+    if (settings::get<bool>("login.MOG_WARDROBE_8"))
+    {
+        mask |= FEATURE_DISPLAY::MOG_WARDROBE_8;
+    }
+
+    return mask;
+}
+
+int32 saveCharacter(uint32 accid, uint32 charid, char_mini* createchar)
+{
+    auto sql = std::make_unique<SqlConnection>();
+
+    const char* Query = "INSERT INTO chars(charid,accid,charname,pos_zone,nation) VALUES(%u,%u,'%s',%u,%u);";
+
+    if (sql->Query(Query, charid, accid, createchar->m_name, createchar->m_zone, createchar->m_nation) == SQL_ERROR)
+    {
+        ShowDebug(fmt::format("lobby_ccsave: char<{}>, accid: {}, charid: {}", createchar->m_name, accid, charid));
+        return -1;
+    }
+
+    Query = "INSERT INTO char_look(charid,face,race,size) VALUES(%u,%u,%u,%u);";
+
+    if (sql->Query(Query, charid, createchar->m_look.face, createchar->m_look.race, createchar->m_look.size) == SQL_ERROR)
```

Comment:
We don't need to be reuding the Query object everywhere  here:

```
if (sql->Query("The query string here",
    charid, createchar->m_look.face, createchar->m_look.race, createchar->m_look.size) == SQL_ERROR)
```

Reply:
I moved most of them that made sense. Some of the larger ones were formatted nicely and I left them alone.